distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1424,54 +1447,244 @@ public void testProductDownloadListEmptyWhenNoDownloads() {
+                .extract().jsonPath().getString("downloadId");
+        // @formatter:on
+        assertThat(downloadId2, not(isEmptyString()));
+
+        String getAllDownloadsUrl = RESOURCE_DOWNLOAD_ENDPOINT_ROOT.getUrl();
+
+        // @formatter:off
+        List<Map<String, Object>> downloads =
+                expect("List of active downloads is not empty").within(30, SECONDS)
+                        .until(() -> when().get(getAllDownloadsUrl).then().log().all()
+                                .extract().body().jsonPath().getList(""), hasSize(2))
+                        .lastResult();
+        // @formatter:on
+
+        Map download1 = downloads.get(0);
+        assertThat(download1.size(), is(6));
+        Map download2 = downloads.get(1);
+        assertThat(download2.size(), is(6));
+
+        if (download1.get("downloadId")
+                .equals(downloadId1)) {
+            checkDownloadInProgress(download1, downloadId1, filename1, resourceData1);
+            checkDownloadInProgress(download2, downloadId2, filename2, resourceData2);
+
+        } else if (download1.get("downloadId")
+                .equals(downloadId2)) {
+            checkDownloadInProgress(download1, downloadId2, filename2, resourceData2);
+            checkDownloadInProgress(download2, downloadId1, filename1, resourceData1);
+        } else {
+            LOGGER.error("Unexpected data in the download");
+            fail("Unexpected data in the download");
+        }
+
+    }
+
+    /**
+     * Determines that when two downloads are downloaded at once and one fails, it does not affect the
+     * success of the other download.
+     * @throws Exception
+     */
+    @Test
+    public void testProductDownloadListWithTwoActiveDownloadsOneFails() throws Exception {
+
+        cometDClient = setupCometDClient(Arrays.asList(NOTIFICATIONS_CHANNEL, ACTIVITIES_CHANNEL));
+
+        String filename1 = "product1.txt";
+        String metacardId1 = generateUniqueMetacardId();
+        String resourceData1 = getResourceData(metacardId1);
+
+        String failFilename = "failProduct.txt";
+        String failMetacardId = generateUniqueMetacardId();
+        String failResourceData = getResourceData(failMetacardId);
+
+        int delayBetweenRetries = 200;
+
+        String startDownloadUrl1 = setUpDownloadResponse(metacardId1, filename1, resourceData1, NO_RETRIES, delayBetweenRetries);
+        String startDownloadUrlFail = setUpDownloadResponse(failMetacardId, failFilename, failResourceData, FAIL_RETRIES, delayBetweenRetries);
+
+        String restUrlFail = REST_PATH.getUrl() + "sources/" + CSW_STUB_SOURCE_ID + "/" + failMetacardId
+                + "?transform=resource" + "&session=" + cometDClient.getClientId();
+
+        // Verify that product retrieval fails from the csw stub server.
+        // @formatter:off
+        when().get(restUrlFail).then().log().all().assertThat().statusCode(500).contentType("text/plain")
```

Comment:
In order for this test to work, we should start two async downloads (L1591-L1601), make sure we get proper the completed and failed notifications, download the resource that succeeded one more time, validate that the expected data is returned and verify that the stub server wasn't called.